src/Microsoft.TestPlatform.CommunicationUtilities/Messages/DiscoveryCompletePayload.cs

```diff
@@ -31,4 +31,19 @@ public class DiscoveryCompletePayload
     /// Gets or sets the Metrics
     /// </summary>
     public IDictionary<string, object> Metrics { get; set; }
+
+    /// <summary>
+    /// Gets or sets list of sources which were fully discovered.
+    /// </summary>
+    public IReadOnlyCollection<string> FullyDiscoveredSources { get; set; } = new List<string>();
```

Comment:
We could make the following 3 properties `get; init;` because we don't really expect them to be set.  But we will need to add manually some type and I am afraid we end up in the same situation as the PR where I try to add shim layer (see https://stackoverflow.com/a/64749403/4625433).